src/Spice86.Core/Emulator/Gdb/GdbCommandBreakPointHandler.cs

```diff
@@ -141,7 +176,30 @@ private void SendS05StringToGdb() {
                 }
                 return null;
             }
-            return new AddressBreakPoint((BreakPointType)breakPointType, address, OnBreakPointReached, false);
+            
+            // Compile condition expression if present
+            Func<long, bool>? condition = null;
+            if (!string.IsNullOrWhiteSpace(conditionExpression) && _state != null && _memory != null) {
+                try {
+                    Shared.Emulator.VM.Breakpoint.Expression.ExpressionParser parser = new();
```

Comment:
Make the parser produce existing data structure IVisitableAstNode.
This can be converted to C# Expression tree which is fast to execute.
Make the breakpoint use that.